File: 096.UniqueBinarySearchTrees/clywin123/numTrees.c
int numTrees(int n) {
	int array[100]={0},i,j,count;
	array[0]=1;
	array[1]=1;
	for(i=2;i<=n;i++){
		count = 0;
		for(j=0;j<i;j++){
			count += (array[j]*array[i-j-1]);
		}
		array[i]=count;
	}
	return array[n];
}

File: 096.UniqueBinarySearchTrees/wyrj/numTrees.js
/**
 * @param {number} n
 * @return {number}
 */
var numTrees = function(n) {
    const map = [1, 1];
    const r = (k) => {
        if (map[k]) return map[k];
        
        map[k] = 0;
        for (let i = 0; i < k; i++) {
            map[k] += r(i) * r(k - i - 1);
        }
        return map[k];
    }
    return r(n);
};
